src/actions/auto-merge.ts

```diff
@@ -1,82 +1,129 @@
-import { info, error, warning } from '../logger';
-import * as github from '../github';
+import { inspect } from 'util';
+import * as core from '@actions/core';
+import * as github from '@actions/github';
+import { Inputs, Strategy, ReviewerBySate, Reviewer } from '../config/typings';
+import { info, debug, warning, error } from '../logger';
+import {
+  getReviewsByGraphQL,
+  removeDuplicateReviewer,
+  filterReviewersByState,
+} from '../github';
 
 export async function run(): Promise<void> {
   try {
-    info('Starting pr auto merge.');
-
-    let config;
-
-    try {
-      config = await github.fetchConfig();
-    } catch (err) {
-      if ((err as Record<string, unknown>).status === 404) {
-        warning(
-          'No configuration file is found in the base branch; terminating the process',
-        );
-        info(JSON.stringify(err));
-        return;
+    info('Staring PR auto merging.');
+
+    let doNotMerge = false;
+
+    const [owner, repo] = core.getInput('repository').split('/');
+
+    const configInput: Inputs = {
+      comment: core.getInput('comment'),
+      owner,
+      repo,
+      pullRequestNumber: Number(core.getInput('pullRequestNumber', { required: true })),
+      sha: core.getInput('sha', { required: true }),
+      strategy: core.getInput('strategy', { required: true }) as Strategy,
+      token: core.getInput('token', { required: true }),
+    };
+
+    debug(`Inputs: ${inspect(configInput)}`);
+
+    const client = github.getOctokit(configInput.token);
+
+    const { data: pullRequest } = await client.pulls.get({
+      owner,
+      repo,
+      pull_number: configInput.pullRequestNumber,
+    });
+
+    info('Checking requested reviewers.');
+
+    if (pullRequest?.requested_reviewers) {
+      const requestedChanges = pullRequest?.requested_reviewers?.map(
+        (reviewer) => reviewer.login,
+      );
+
+      debug(JSON.stringify(requestedChanges, null, 2));
+
+      if (requestedChanges.length > 0) {
+        warning(`Waiting [${requestedChanges.join(', ')}] to approve.`);
+        doNotMerge = true;
```

Comment:
We can have a case.
Approvers:
- user1
- user2 (required)

user1 request changes
PR is not merged.
user2 approves

The author adds +1 commit.
user1, user2 reassigned, now they are both pending state.

user2 approves.
user1 had previously requested changes, but now he didn't re-review changes.
PR gets merged